### PR TITLE
added communication with REMS-Admin

### PR DESCRIPTION
**Pull request by smalho01**
https://github.com/mcode/REMS/pull/17, https://github.com/mcode/dtr/pull/22, and https://github.com/mcode/CRD/pull/19 are also a part of this PR, both of those should be checked out and reviewed alongside this one.

To see communication between PIS and REMS Admin, follow the REMS workflow steps as normally. Once the questionnaire is filled out at the end of DTR, hitting the submit button will send the compliance bundle to the rems admin, receive a response with the UID case number and status, and then forward that response to the PIS system. Exactly a minute after the submit button is hit, the status of the request will change from "Pending" to "Approved", mocking/simulating REMS admin intervention. This can be seen in the PIS system by hitting the "Order Verify" button under new doctor orders (will probably have to zoom out to see this button). Hitting this button after one minute will see the order move over to the verified doctor orders tab (you have to switch tabs for it to update, the state doesn't automatically update unfortunately). Before one minute and the order will stay under the new doctor orders tab.

### CODE REVIEW

**Review by KeeyanGhoreshi — COMMENTED** [review]
Is all the commented out code still needed in some way?  Or can we just delete it?

**Comment by smalho01** [comment]
> Is all the commented out code still needed in some way? Or can we just delete it?

All the commented out code shouldn't really be needed for anything dealing with the demo of the prototype, however I decided to comment it out instead of straight up deleting it incase referencing some of the legacy code helps out with other parts of the PIS in future work, ie. stuff that's not going to be in the REMS v0.3 demo. I think we should finish going through the PIS system code clean up task before we delete large chunks of code but we certainly can once we're more confident in the state of the PIS code base.

File: backend/routes/doctorOrders.js [hunk 1]
```diff
@@ -270,6 +250,104 @@ async function sendMail(user, callback) {
   callback(info);
 }
 
+function parseRemsAdminRequest(requestBody , existingDocOrder = undefined) {
```
Comment (KeeyanGhoreshi) [inline backend/routes/doctorOrders.js:253]: This function could use some null-checking, I don't think the resources we use are profiled in any way to ensure that there is a family/given name, for example.

Good place to use the some [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).

File: src/app/mainwindow/a-inventory-window/a-shopping-cart-window/DoctorOrderServices.service.ts [hunk 2]
```diff
@@ -40,19 +40,8 @@ export class DoctorOrderServices{
...
-                            drugId:drugId,
-                            drugName:drugName ,
-                            drugPrice:drugPrice,
-                            drugQuantity:drugQuantity,
-                            realQuantity:realQuantity,
-                            totalAmount:totalAmount,
-                            pickupDate:pickupDate};
-    this.http.post(environment.backendBaseUrl + "/api/verifiedDoctorOrder",VerifiedDoctorOrderData)
+  createVerifiedDoctorOrder(id: string){
+    this.http.patch(environment.backendBaseUrl + "/api/doctorOrder/fhir/rems/" + id, {})
```
Comment (KeeyanGhoreshi) [inline src/app/mainwindow/a-inventory-window/a-shopping-cart-window/DoctorOrderServices.service.ts]: minor comment, but we're using `HttpClient` here to make requests, but use `axios` elsewhere.  We should try to stay consistent throughout the project.

**Review by KeeyanGhoreshi — COMMENTED** [review]
overall it works, just two minor comments.

File: src/app/mainwindow/a-inventory-window/a-shopping-cart-window/DoctorOrderServices.service.ts (hunk 2 above)
Comment (smalho01) [reply src/app/mainwindow/a-inventory-window/a-shopping-cart-window/DoctorOrderServices.service.ts]: HttpClient is angular's defualt package for network requests and is used a lot in the frontend and should probably keep being used. The backend however is run within node and is where the axios commands exist. I'm not sure if the backend would work well with HttpClient. If you look at the backend as separate from the frontend and consider them to be two separate projects even though they are in the same repo, then we are fairly consistent with which we use I think.

File: backend/routes/doctorOrders.js (hunk 1 above)
Comment (smalho01) [reply backend/routes/doctorOrders.js:253]: yeah I can try adding some null checking stuff here

File: src/app/mainwindow/a-inventory-window/a-shopping-cart-window/DoctorOrderServices.service.ts (hunk 2 above)
Comment (KeeyanGhoreshi) [reply src/app/mainwindow/a-inventory-window/a-shopping-cart-window/DoctorOrderServices.service.ts]: makes sense

**Review by KeeyanGhoreshi — APPROVED** [review]

**Pull request merged**